quic/client/handshake/ClientHandshake.cpp

```diff
@@ -19,15 +20,41 @@ ClientHandshake::ClientHandshake(QuicClientConnectionState* conn)
 
 void ClientHandshake::connect(
     folly::Optional<std::string> hostname,
-    folly::Optional<fizz::client::CachedPsk> cachedPsk,
+    folly::Optional<QuicCachedPsk> quicCachedPsk,
     std::shared_ptr<ClientTransportParametersExtension> transportParams,
     HandshakeCallback* callback) {
   transportParams_ = std::move(transportParams);
   callback_ = callback;
 
+  folly::Optional<fizz::client::CachedPsk> cachedPsk;
+  if (quicCachedPsk) {
+    cachedPsk = std::move(quicCachedPsk->cachedPsk);
+  }
+
   connectImpl(std::move(hostname), std::move(cachedPsk));
 
   throwOnError();
+
+  if (conn_->zeroRttWriteCipher) {
+    if (conn_->qLogger) {
+      conn_->qLogger->addTransportStateUpdate(kZeroRttAttempted);
+    }
+    QUIC_TRACE(zero_rtt, *conn_, "attempted");
+
+    // If zero rtt write cipher is derived, it means the cached psk was valid
+    DCHECK(quicCachedPsk);
+
+    auto& transportParams = quicCachedPsk->transportParams;
```

Comment:
This causes a shadowing warning because of the name.

Reply:
You got it.